packages/firebase_crashlytics/firebase_crashlytics_platform_interface/lib/src/method_channel/method_channel_crashlytics.dart

```diff
@@ -87,9 +90,35 @@ class MethodChannelFirebaseCrashlytics extends FirebaseCrashlyticsPlatform {
     required String exception,
     required String information,
     required String reason,
+    bool fatal = false,
     List<Map<String, String>>? stackTraceElements,
   }) async {
     try {
+      /// "fatal" is an optional parameter that we're using to signal to the Crashlytic's service that this particular
+      /// error was a fatal one.  The below if statement is Firebase's prescribed method of signalling a fatal error.
+      if (fatal) {
+        try {
+          num currentUnixTimeSeconds =
+              (DateTime.now().millisecondsSinceEpoch / 1000).ceil();
+
+          await setCustomKey('com.firebase.crashlytics.flutter.fatal',
```

Comment:
There is still the very-small time window in any given rounded-up second where this `await setCustomKey` may take long enough to roll into the next second, at which point when the error is recorded it will not meet the requirement of "crash timestamp that is closest to but not greater than the custom key timestamp". As a guess it will probably happen on order of ever 1,000 to 100,000 times (depending on setCustomKey latency? I'm unaware of it's latency) meaning it will skew statistics mostly and likely never ever be really detected, but it will be there

Reply:
(in RNFB and the upstream example they just added a second to the calculation...)